yellowbrick/classifier/class_balance_heat_map.py

```diff
@@ -0,0 +1,169 @@
+# yellowbrick.classifier.class_balance_heat_map
+# Class balance visualizer for showing per-class support.
+#
+# Author:   Rebecca Bilbro <rbilbro@districtdatalabs.com>
+# Author:   Benjamin Bengfort <bbengfort@districtdatalabs.com>
+# Author:   Neal Humphrey
+# Author:   Larry Gray
+# Created:  Sun March 4 20:45:00 2018 -0400
+#
+# Copyright (C) 2017 District Data Labs
+# For license information, see LICENSE.txt
+#
+# ID: class_balance_heat_map.py [5388065] lwgray@gmail.com $
+
+"""
+Class balance heat map visualizer for showing per-class support.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import numpy as np
+
+from .base import ClassificationScoreVisualizer
+
+from sklearn.model_selection import train_test_split
+from sklearn.utils.multiclass import unique_labels
+
+
+##########################################################################
+## Class Balance Heat Map Chart
+##########################################################################
+
+class ClassBalanceHeatMap(ClassificationScoreVisualizer):
+    """
+    Class balance heat map that shows the support for each class in the
+    fitted classification model displayed as a stacked bar.  Each bar is
+    segmented to show the distribution of predicted classes for each
+    class. It is initialized with a fitted model and generates a
+    class balance heat map chart on draw.
+
+    Parameters
+    ----------
+    ax: axes
+        the axis to plot the figure on.
+    model: estimator
+        Scikit-Learn estimator object. Should be an instance of a classifier,
+        else ``__init__()`` will raise an exception.
+    classes: list
+        A list of class names for the legend. If classes is None and a y value
+        is passed to fit then the classes are selected from the target vector.
+    kwargs: dict
+        Keyword arguments passed to the super class. Here, used
+        to colorize the bars in the histogram.
+    Notes
+    -----
+    These parameters can be influenced later on in the visualization
+    process, but can and should be set as early as possible.
+    """
+
+    def score(self, X, y=None, **kwargs):
```

Comment:
As part of the discussion, one thing I will say is that I think `ClassBalance` is actually a feature visualizer, or maybe something new... a target visualizer! Having `score` as the entry point always seemed a little too far downstream, as did involving a model, since knowledge of the class balance is usually a preliminary to selecting a model. 

This version, however, is definitely a `ModelVisualizer` and its entry point is definitely `score` (though you could also make a case for `predict`.

Reply:
I agree, I found it weird that I couldn't get class information until the very end of my workflow.